modules/snooker/src/main/ts/ephox/snooker/operate/Redraw.ts

```diff
@@ -80,13 +83,20 @@ const render = function <T extends DetailNew> (table: SugarElement, grid: RowDat
       case 'tfoot':
         footSection.push(row);
         break;
+      case 'colgroup':
+        columnGroupsSection.push(row);
+        break;
     }
   });
 
   renderOrRemoveSection(headSection, 'thead');
   renderOrRemoveSection(bodySection, 'tbody');
   renderOrRemoveSection(footSection, 'tfoot');
 
+  if (columnGroupsSection.length) {
```

Comment:
This will incorrectly render the table in some cases, as `colgroup` must be **before** the `thead` element, but after the `caption`. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table

Reply:
This was only half fixed, so I'm reopening this comment. You also need to ensure `renderOrRemoveSection` places it correctly if it's been moved (similar to the `insertThead` logic).